intelmq/bots/parsers/alienvault/parser_otx.py

```diff
@@ -33,50 +35,54 @@ def process(self):
+                            else 'http://' + indicator["indicator"]
+                        path = parse.urlparse(resource).path
+                        if len(path) > 0:
+                            event.add('source.url', resource)
+                        else:
+                            event.add('source.fqdn',
+                                      indicator["indicator"])
+                    # IP addresses
+                    elif indicator["type"] in ['IPv4', 'IPv6']:
+                        event.add('source.ip',
+                                  indicator["indicator"])
+                    # emails
+                    elif indicator["type"] == 'email':
+                        event.add('source.account',
+                                  indicator["indicator"])
+                    # URLs
+                    elif indicator["type"] in ['URL', 'URI']:
+                        resource = indicator["indicator"] \
+                            if '://' in indicator["indicator"] \
+                            else 'http://' + indicator["indicator"]
                         event.add('source.url', resource)
-                    else:
-                        event.add('source.fqdn',
+                    # CIDR
+                    elif indicator["type"] in ['CIDR']:
+                        event.add('source.network',
                                   indicator["indicator"])
-                # IP addresses
-                elif indicator["type"] in ['IPv4', 'IPv6']:
-                    event.add('source.ip',
-                              indicator["indicator"])
-                # emails
-                elif indicator["type"] == 'email':
-                    event.add('source.account',
-                              indicator["indicator"])
-                # URLs
-                elif indicator["type"] in ['URL', 'URI']:
-                    resource = indicator["indicator"] \
-                        if '://' in indicator["indicator"] \
-                        else 'http://' + indicator["indicator"]
-                    event.add('source.url', resource)
-                # CIDR
-                elif indicator["type"] in ['CIDR']:
-                    event.add('source.network',
-                              indicator["indicator"])
-
-                # CVE
-                elif indicator["type"] in ['CVE']:
-                    additional_indicator['CVE'] = indicator["indicator"]
-                    # TODO: Process these IoCs: FilePath, Mutex
-                else:
+
+                    # CVE
+                    elif indicator["type"] in ['CVE']:
+                        additional_indicator['CVE'] = indicator["indicator"]
+                        # TODO: Process these IoCs: FilePath, Mutex
+                    else:
+                        continue
+                except InvalidValue:
+                    self.logger.exception('Failed to parse line: {}\n{}'.format(indicator, format_exc()))
```

Comment:
When using `logger.exception` the traceback is appended automatically: https://docs.python.org/3/library/logging.html#logging.Logger.exception

But AFAIS the message is not dumped at all then, it's only logged. Which means you need to get the message from the logs, add it to the dump file and then recover it. That's neither intuitive nor in line with all the other bots.